anaconda_project/project_commands.py

```diff
@@ -124,6 +126,35 @@ def add_args(self, results, args):
         return added + args
 
 
+class _TemplateArgsTransformer(_ArgsTransformer):
+    """ArgsTransformer that supports jinja2 templating."""
+
+    def add_args(self, results, args):
+        return args
+
+    def arg_to_identifier(self, arg):
+        """Turn a commandline argument into a Python identifier for jinja2."""
+        return arg.replace('--','').replace('-','_')
+
+    def parse_and_template(self, command, extra_args):
+        results = {spec.option: [] for spec in self.specs}
+        self._parse_args_removing_known(results, extra_args)
+        extra_args = _TemplateArgsTransformer().transform_args(extra_args)
+        _append_extra_args_to_command_line(command, extra_args)
+        items = {self.arg_to_identifier(k):(v[0] if v else True)
+                 for k,v in results.items() if v}
+        replacements = {}
+        for k,v in items.items():
+            if k.startswith('anaconda_project_'):
+                replacement = k.replace('anaconda_project_', '')
+                if replacement not in items:
+                    replacements[replacement] = v
```

Comment:
If I'm reading this correctly, it seems like `--anaconda-project-port 5000 --anaconda-project-port 6000` would result in `{"port":5000}`? I'd prefer `{"port":6000}` in such a case, reading left to right, and doing so seems like it would slightly simplify the code (removing an `if`).

Reply:
@jbednar Running``--anaconda-project-port 5000 --anaconda-project-port 6000`` currently *does* result in `{"port":5000}` but it has nothing to do with this code. 

Instead it is due to this line `items = {k: (v[0] if v else True) for k, v in results.items() if v}` because the existing parser parses such a command  into ``{'--anaconda-project-host': [], ...,  '--anaconda-project-port': ['5000', '6000']}``(now we know why it was a list!). The fix is to index the last item in the list with `v[-1]` which is what I did in c64d7d0.